src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -260,18 +258,23 @@ bool java_bytecode_languaget::typecheck(
     get_message_handler());
 
   // now typecheck all
-  auto res = java_bytecode_typecheck(
-       symbol_table, get_message_handler(), string_refinement_enabled);
-
-  generate_java_generic_typet new_symbol_generator(symbol_table, get_message_handler());
+  bool res=java_bytecode_typecheck(symbol_table, get_message_handler(), string_refinement_enabled);
+//  if (!res)
```

Comment:
I think this just need to be toggled (that is `java_bytecode_typecheck` returns `false` when everything went well)